homeassistant/components/sonos/number.py

```diff
@@ -0,0 +1,65 @@
+"""Entity representing a Sonos number control."""
+from __future__ import annotations
+
+from homeassistant.components.number import NumberEntity
+from homeassistant.const import ENTITY_CATEGORY_CONFIG
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from .const import SONOS_CREATE_LEVELS
+from .entity import SonosEntity
+from .helpers import soco_error
+from .speaker import SonosSpeaker
+
+LEVEL_TYPES = ("bass", "treble")
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Sonos number platform from a config entry."""
+
+    async def _async_create_entities(speaker: SonosSpeaker) -> None:
+        entities = []
+        for level_type in LEVEL_TYPES:
+            entities.append(SonosLevelEntity(speaker, level_type))
+        async_add_entities(entities)
+
+    config_entry.async_on_unload(
+        async_dispatcher_connect(hass, SONOS_CREATE_LEVELS, _async_create_entities)
+    )
+
+
+class SonosLevelEntity(SonosEntity, NumberEntity):
+    """Representation of a Sonos level entity."""
+
+    _attr_entity_category = ENTITY_CATEGORY_CONFIG
+    _attr_entity_registry_enabled_default = False
```

Comment:
They have the config entity category... why disable them?

Reply:
I put this in as a last-second change before submitting. Seemed like things were getting a bit noisy and I'd expect these are not commonly used. If it seems reasonable to leave them enabled I can remove this.

Reply:
I'm not sure why they are disabled? Is there a technical reason?

Reply:
No technical reason, just hiding uncommonly used entities.

Reply:
But the entity category already hides the non-primary enties?

Reply:
It does. Still getting used to the new freedom to add extra entities without worry. 😄